actions/dump-charm-artifacts/action.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Dump Charm Log Artifacts'
+description: 'Dumps to Artifacts useful Juju, Kubernetes, and Charmcraft log information.  Requires kubectl to be set up with a context, and charmcraft to be installed.'
```

Comment:
why cannot we install kubectl and charmcraft?

eg we check if they are present, if not install

Reply:
Yeah I think what @beliaev-maksim you mention is technically possible, but @DnPlas's point about context is good (we don't really know how to initialise kubectl here).  I worry a bit about clobbering existing installs too (if we overwrite the existing charmcraft or kubectl), but we could do checks for that.

Reply:
I lean toward having these as dependencies rather than something set up by the action just because it is a bit odd to use this somewhere that you don't have kubectl and charmcraft already.  But am open to changing it

Reply:
@ca-scribner can we have at least a step (ideally in bash script) that checks all prerequisites ?

Reply:
I think it's alright to keep them as deps, as for checking pre-requisites, I agree with @beliaev-maksim

Reply:
added.  tested locally and it works, but calling all bash experts to check my math :)

Reply:
yes, prereqs are good 

I tested locally and looks to work